### DISCUSSION TITLE
Property Presets for common mod types

### DISCUSSION THREAD

**Discussion by StjerneIdioten** [StjerneIdioten/I3D-Blender-Addon#109] [Ideas & Feature Requests]
**Is your feature request related to a problem? Please describe.**
It is annoying to setup all of the properties every time for common scenarios.

**Describe the solution you'd like**
A solution similar to the way the maya addon does it with presets for different mod types

**Additional context**
![image](https://user-images.githubusercontent.com/11218511/76127413-df8cc580-6001-11ea-898c-efddfe790175.png)

![image](https://user-images.githubusercontent.com/11218511/76127460-fdf2c100-6001-11ea-8f4f-f948b37aa1a6.png)

![image](https://user-images.githubusercontent.com/11218511/76127472-077c2900-6002-11ea-90b2-b6944fb65037.png)

**Comment by Kieruss** [comment]
This would be great, especially if we can create our own presets (even if we have to edit an xml file). Apart from having defaults for newly created materials, I assume this would allow us to apply a preset/template to an existing material? If the template/preset could be configured in such a way that each option can either be forced, only if empty, or ignored/no-change, that would be great. That would, for example, allow me to apply the "default maps" template, which would assign the default normal, and specular images, without changing anything else (such as if I had a custom diffuse set). In that way, we could layer templates to quickly get the setup we want.

**Reply by StjerneIdioten** [reply]
I mean sure, that wouldn't be a problem to program as far as I can tell. It would involve some custom menu of course, because I can't really inserts buttons etc. into the existing material menu. But I was planning on having that for the template stuff anyway as well as making all of the addon/exporter parameters user customizable. So you can set your own "defaults" for the i3d stuff if you like to (Like the ever requested "shadows" to be on by default) and have your preferred exporter parameters every time you create a new project, since currently you have to set them on a per .blend file basis.
